src/profile-logic/transforms.js

```diff
@@ -905,6 +937,76 @@ const FUNC_MATCHES = {
   },
 };
 
+export function collapseFunctionSubtree(
+  thread: Thread,
+  funcToCollapse: IndexIntoFuncTable
+): Thread {
+  const { stackTable, frameTable, samples } = thread;
+  const oldStackToNewStack: Map<
+    IndexIntoStackTable | null,
+    IndexIntoStackTable | null
+  > = new Map();
+  oldStackToNewStack.set(null, null);
+  const collapsedStacks = new Set();
+  const newStackTable = {
+    length: 0,
+    prefix: [],
+    frame: [],
+  };
+
+  for (let stackIndex = 0; stackIndex < stackTable.length; stackIndex++) {
+    const prefix = stackTable.prefix[stackIndex];
+    const frameIndex = stackTable.frame[stackIndex];
+    const funcIndex = frameTable.func[frameIndex];
```

Comment:
nit: we use frameIndex and funcIndex only in the `else` block, could you please move these 2 lines there? Having them closer to their use will make it easier to follow.

Reply:
I moved them both to be nearest to their first use.